pyccel/codegen/printing/ccode.py

```diff
@@ -693,20 +688,18 @@ def _print_NumpyRandint(self, expr):
 
     def _print_FunctionDef(self, expr):
 
+        if len(expr.results) > 1:
+            self._additional_args = expr.results
```

Comment:
This will not work in the case of a function in a function. I suggest using a list of lists so that this line becomes:
`self._additional_args.append(expr.results)`